posthog/api/test/test_capture.py

```diff
@@ -143,6 +162,148 @@ def test_capture_event(self, kafka_produce):
             self._to_arguments(kafka_produce),
         )
 
+    @patch("posthog.kafka_client.client._KafkaProducer.produce")
+    def test_capture_event_with_db_error_no_default(self, kafka_produce):
+        """Test database errors while fetching instance settings don't fail capture.
+
+        Even though we are creating an InstanceSetting, it should not be fetched as we
+        force a database error to occur (with the DatabaseErrorWrapper). This should
+        cause us to use the default EVENT_PARTITION_KEYS_TO_OVERRIDE setting. As this
+        is not set to the current expected_key, no override will occur.
+        """
+        distinct_id = "999"
+        expected_key = f"{self.team.pk}:{distinct_id}"
+        key = CONSTANCE_DATABASE_PREFIX + "EVENT_PARTITION_KEYS_TO_OVERRIDE"
+
+        InstanceSetting.objects.create(key=key, raw_value=f'["{expected_key}"]')
+        # Assume no cache for this test
+        cache.clear()
+
+        data = {
+            "event": "$autocapture",
+            "properties": {
+                "distinct_id": distinct_id,
+                "token": self.team.api_token,
+                "$elements": [
+                    {"tag_name": "a", "nth_child": 1, "nth_of_type": 2, "attr__class": "btn btn-sm"},
+                    {"tag_name": "div", "nth_child": 1, "nth_of_type": 2, "$el_text": "💻"},
+                ],
+            },
+        }
+        patched_config = {"EVENT_PARTITION_KEYS_TO_OVERRIDE": ("not_current_expected_key", "a help str", str)}
+        with patch.dict(CONSTANCE_CONFIG, patched_config, clear=True):
+            with simulate_postgres_error("posthog_instancesetting"):
```

Comment:
when postgres goes down, it's not just this one table that's down but all of them, so this _seems_ ineffective to test to me. What was the intention here?

Reply:
We are not testing whether Postgres is down, we are testing whether our query to `posthog_instancesetting` fails (for whatever reason). Postgres may still be up even if one query fails.